src/i18n.js

```diff
@@ -1,32 +1,41 @@
 import i18n from 'i18next';
-import { reactI18nextModule } from 'react-i18next';
-import LanguageDetector from 'i18next-browser-languagedetector';
```

Comment:
This is a bit controversial. The plugin detects the language not only based on the browser but from all kind of sources (query strings, local storage, etc). From one point of view, this makes the component very "fragile" as the language can "unexpectedly" change. It seems that this is something to avoid.

The contra argument is that using this plugin can "automatically" provide the "right" language based on the app where the component is used. It does seem like a valid point to some extent. If we decide to take this approach, we would have to remove the "locale" prop as it makes the props and actual language inconsistent. 

I feel that using a prop where the client can explicitly state what language the component should use is slightly better. Thoughts?

Reply:
I would go in this order:
1. The host UI gets the locale from customizr and passes it to the component.
2. The host UI detects the lang in another way (lang detector) and passes it.
3. The component detects the locale (customizr, then lang detector if there is no conf).

Reply:
so:
1. and 2. means - rely on the props (from a component point of view). 👍 
3. feels a bit too much of calls to make from this component. I feel the customizr might be better called at app level (once) and then the data should be injected to each component (as there might be quite a few)

Reply:
I'm sorry, I failed to properly communicate what I meant:
The idea was that we'll try to do 1 and if it fails (no data in Customizr) we'll fall back to 2. 3 is the last fallback which is hit when the host UI doesn't pass the locale to the component via a prop. So, it's basically having all 1, 2, and 3 working together to ensure we're going to display the right language.

The part I'm not sure about is the case when we have a UI which detects some language (e.g. English) but fails to pass it to the component. The component looks at the data in Customizr and gets another language (say Chinese). Now we have a UI in a mix of languages (all the text is in Englis but one component is in Chinese).